### ISSUE TITLE
Why stage training?

### ISSUE THREAD

**Issue by ZhengMengbin** [xyfJASON/HCL#2]
What's the differences between first phase training, second phase training and downstream tasks fine-tuning?

**Comment by xyfJASON** [comment]
In the first phase (before the convergence of contrastive learning), we feed the **ground-truth masks** to the decoder, given that the predicted masks aren't sufficiently accurate at this point. In the second phase (after the convergence of contrastive learning), we switch to the **masks predicted by our hierarchical contrastive learning**.
For the downstream tasks (watermark removal and shadow removal in our paper), we fine-tune the model based on the weights pretrained on Places dataset.

**Comment by xyfJASON** [comment]
Closing this issue due to inactivity. Feel free to reopen if you have more questions.
